### PR TITLE
add recipe for shannon RNAseq assembler

### PR DESCRIPTION
**Pull request by bluegenes**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Shannon RNAseq assembler: http://sreeramkannan.github.io/Shannon/about
Note: shannon.patch adds a shebang to shannon.py. We also addressed this issue via PR to Shannon: https://github.com/sreeramkannan/Shannon/pull/26

### CODE REVIEW

File: recipes/shannon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+source:
+  url: https://github.com/sreeramkannan/Shannon/archive/{{ version }}.tar.gz
+  sha256: b089778da922233f695e01e2d2b7dc05a6c2f9f2328fa793d163421e75cf44ce
+  patches:
+    - shannon.patch
+
+build:
+  number: 0
+  noarch: generic
```
Comment (bgruening) [inline recipes/shannon/meta.yaml]: I guess this pkgs could be "noarch: python"?

File: recipes/shannon/shannon.patch [hunk 2]
```diff
@@ -0,0 +1,16 @@
+--- shannon.py.bak	2018-08-30 11:47:27.138347197 -0700
++++ shannon.py	2018-08-30 12:01:56.090925724 -0700
+@@ -1,3 +1,4 @@
++#! /usr/bin/env python2
```
Comment (bgruening) [inline recipes/shannon/shannon.patch]: just python is enough

**Comment by bluegenes** [comment]
Thanks @bgruening - changes made!

**Comment by bgruening** [comment]
Cool, thanks! Lets us know if you consider Shannon production ready to replace Trinity :)

**Pull request merged**